scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/job/termination/handlers/TerminateLoopHandler.java

```diff
@@ -0,0 +1,54 @@
+package org.ow2.proactive.scheduler.job.termination.handlers;
+
+import java.util.Date;
+
+import org.apache.log4j.Logger;
+import org.ow2.proactive.scheduler.common.task.TaskId;
+import org.ow2.proactive.scheduler.common.task.flow.FlowAction;
+import org.ow2.proactive.scheduler.core.SchedulerStateUpdate;
+import org.ow2.proactive.scheduler.job.ChangedTasksInfo;
+import org.ow2.proactive.scheduler.job.InternalJob;
+import org.ow2.proactive.scheduler.task.internal.InternalTask;
+import org.ow2.proactive.scheduler.util.policy.ISO8601DateUtil;
+
+import it.sauronsoftware.cron4j.InvalidPatternException;
+import it.sauronsoftware.cron4j.Predictor;
+
+public class TerminateLoopHandler {
+
+	public static final Logger logger = Logger.getLogger(TerminateLoopHandler.class);
+
+	private final InternalJob internalJob;
+
+	public TerminateLoopHandler(InternalJob internalJob) {
+		this.internalJob = internalJob;
+	}
+
+	public boolean terminateLoopTask(boolean didAction, FlowAction action, InternalTask initiator,
+			ChangedTasksInfo changesInfo, SchedulerStateUpdate frontend) {
+		// find the target of the loop
+		InternalTask target = null;
+		if (action.getTarget().equals(initiator.getName())) {
+			target = initiator;
+		} else {
+			target = internalJob.findTaskUp(action.getTarget(), initiator);
+		}
+		didAction = internalJob.replicateForNextLoopIteration(initiator, target, changesInfo, frontend, action);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Introduce a new variable instead of reusing the parameter "didAction". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1226)